src/content/en/updates/posts/2016/03/web-enabled-internet-of-things.markdown

```diff
@@ -0,0 +1,523 @@
+is much cooler than bringing your own inventions to life, and
+being able to talk to them!
+</p>
+
+![Client application](/web/updates/images/2016/03/web-enabled-internet-of-things/tablet-ux.png)
+
+# Web and IoT, a match to be
+
+There are still a lot of hurdles to be overcome before Internet
+of Things will be a huge success. Personally, I am worrying about
+companies requiring people to install their apps for each device
+they produce, cluttering my phone with a multitude of apps that
+I rarely use.
+
+For this reason, I have been very excited about the Physical Web
+project, which allows devices to broadcast a URL to an online
+website, in a non-intrusive way. In combinations with emerging
+web technologies such as Web Bluetooth, Web USB and Web NFC, the
+site can either connect directly to the device or at least explain
+the proper way of doing so.
+
+Web Bluetooth would be a good way of doing so, but in some cases
+you might require a physical connection (say via USB) for security
+reasons, for instance in an office setting.
+
+The web site can also serve as what is called a Progressive Web
+App (aka PWA), which means that the site has a responsive app like
+user experience, works offline and can be added to the home screen.
+Google has a really good explainer about PWAs, which I urge you to
+check out:
+
+[Progressive Web Apps](https://developers.google.com/web/progressive-web-apps)
+
+As a proof of concept, I have been building a small device using
+the Intel&reg; Edison Arduino breakout board. It contains a temperature
+sensor (TMP36) as well as an actuator (colored LED cathode).
+
+![Breadboard](/web/updates/images/2016/03/web-enabled-internet-of-things/breadboard.jpg)
+
+The Edison is an interesting device as it can runs a full Linux
+distribution and therefore allows me to program it easily using
+Node.JS. The installer allows for installing the Intel&reg; XDK which
+makes it easy to get started, though you can program and upload
+to your device manually as well.
+
+> NOTE: It is possible to use Brillo or Ostro instead of the
+> default software, in which case you should follow the documentation
+> for these OSes for now to get a Node.JS application running on the
+> device.
+
+For my Node.JS app, I required three node modules, as well as their
+dependencies: `eddystone-beacon`, `parse-color`, `johnny-five`. The
+former will automatically install `noble` which is the node module
+that I use for talking Bluetooth Low Energy. 
+
+> NOTE: It is important to not list `noble` as a dependency in the
+> `package.json` file, as you need to use the same noble instance as
+> `eddystone-beacon`, in order for them to work together.
+
+You can find more info [here](https://github.com/don/node-eddystone-beacon/issues/30)
+
+The `package.json` file for the project looks like this:
+
+{% highlight javascript %}
+{
+  "name": "edison-webbluetooth-demo-server",
+  "version": "1.0.0",
+  "main": "main.js",
+  "engines": {
+    "node": ">=0.10.0"
```

Comment:
The default software works with 0.10 but if you install the brillo or ostro os, you will get something newer (at least with ostro but heard it is the same with brillo)